modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/ghoul.dm

```diff
@@ -0,0 +1,282 @@
+	if (chest?.type != part_default_chest)
+		qdel(chest)
+		var/obj/item/bodypart/limb = new part_default_chest
+		limb.replace_limb(src,TRUE)
+	var/obj/item/bodypart/l_arm = get_bodypart(BODY_ZONE_L_ARM)
+	if (l_arm?.type != part_default_l_arm)
+		qdel(l_arm)
+		var/obj/item/bodypart/limb = new part_default_l_arm
+		limb.replace_limb(src,TRUE)
+	var/obj/item/bodypart/r_arm = get_bodypart(BODY_ZONE_R_ARM)
+	if (r_arm?.type != part_default_r_arm)
+		qdel(r_arm)
+		var/obj/item/bodypart/limb = new part_default_r_arm
+		limb.replace_limb(src,TRUE)
+	var/obj/item/bodypart/l_leg = get_bodypart(BODY_ZONE_L_LEG)
+	if (l_leg?.type != part_default_l_leg)
+		qdel(l_leg)
+		var/obj/item/bodypart/limb = new part_default_l_leg
+		limb.replace_limb(src,TRUE)
+	var/obj/item/bodypart/r_leg = get_bodypart(BODY_ZONE_R_LEG)
+	if (r_leg?.type != part_default_r_leg)
+		qdel(r_leg)
+		var/obj/item/bodypart/limb = new part_default_r_leg
+		limb.replace_limb(src,TRUE)
+
+/datum/species/ghoul/on_species_loss(mob/living/carbon/human/C, datum/species/new_species, pref_load)
+	..()
+
+	// 2) BODYPARTS
+	C.part_default_head = /obj/item/bodypart/head
+	C.part_default_chest = /obj/item/bodypart/chest
+	C.part_default_l_arm = /obj/item/bodypart/l_arm
+	C.part_default_r_arm = /obj/item/bodypart/r_arm
+	C.part_default_l_leg = /obj/item/bodypart/l_leg
+	C.part_default_r_leg = /obj/item/bodypart/r_leg
+	C.ReassignForeignBodyparts()
+
+//////////////////
+// ATTACK PROCS //
+//////////////////
+
+/datum/species/ghoul/disarm(mob/living/carbon/human/user, mob/living/carbon/human/target, datum/martial_art/attacker_style)
+	// Targeting Self? With "DISARM"
+	if (user == target)
+		var/target_zone = user.zone_selected
+		var/list/allowedList = list ( BODY_ZONE_L_ARM, BODY_ZONE_R_ARM, BODY_ZONE_L_LEG, BODY_ZONE_R_LEG )
+		var/obj/item/bodypart/affecting = user.get_bodypart(check_zone(user.zone_selected)) //stabbing yourself always hits the right target
+
+		if ((target_zone in allowedList) && affecting)
+
+			if (user.handcuffed)
+				to_chat(user, span_alert("You can't get a good enough grip with your hands bound."))
+				return FALSE
+
+			// Robot Arms Fail
+			if (affecting.status != BODYPART_ORGANIC)
+				to_chat(user, "That thing is on there good. It's not coming off with a gentle tug.")
+				return FALSE
+
+			// Pry it off...
+			user.visible_message("[user] grabs onto [p_their()] own [affecting.name] and pulls.", span_notice("You grab hold of your [affecting.name] and yank hard."))
+			if (!do_mob(user,target))
+				return TRUE
+
+			user.visible_message("[user]'s [affecting.name] comes right off in their hand.", span_notice("Your [affecting.name] pops right off."))
+			playsound(get_turf(user), 'sound/effects/meatslap.ogg', 40, 1) //ill change these sounds later
+
+			// Destroy Limb, Drop Meat, Pick Up
+			var/obj/item/I = affecting.drop_limb()
+			if (istype(I, /obj/item/food/meat/slab))
+				user.put_in_hands(I)
+
+			return TRUE
+	return ..()
```

Comment:
This should probably be doing some form of damage given you are literally ripping off your own limb